src/main/kotlin/ch/hevs/cloudio/cloud/apiutils/EndpointManagementUtil.kt

```diff
@@ -21,51 +20,60 @@ object  EndpointManagementUtil{
 
     fun createEndpoint(endpointParametersRepository: EndpointParametersRepository, endpointCreateRequest: EndpointCreateRequest): EndpointParameters {
         val toReturn = EndpointParameters(UUID.randomUUID().toString(), endpointCreateRequest.endpointFriendlyName)
-
+        //create endpoint in endpoint parameters repo
         endpointParametersRepository.save(toReturn)
-
         return toReturn
     }
 
-    fun getEndpoint(endpointEntityRepository: EndpointEntityRepository, endpointRequest: EndpointRequest): EndpointEntity? {
-        return endpointEntityRepository.findByIdOrNull(endpointRequest.endpointUuid)
+    fun getEndpoint(endpointEntityRepository: EndpointEntityRepository, endpointParametersRepository: EndpointParametersRepository, endpointRequest: EndpointRequest): EndpointAnswer? {
+        val endpointEntity = endpointEntityRepository.findByIdOrNull(endpointRequest.endpointUuid)
+        return if(endpointEntity == null)
+            null
+        else    //return endpoint entity and endpoint friendly name
+            EndpointAnswer(endpointParametersRepository.findById(endpointRequest.endpointUuid).get().friendlyName, endpointEntity)
+    }
+
+    fun getEndpointFriendlyName(endpointParametersRepository: EndpointParametersRepository, endpointRequest: EndpointRequest): EndpointFriendlyName?{
+        val endpointParameters = endpointParametersRepository.findByIdOrNull(endpointRequest.endpointUuid)
+        return if(endpointParameters == null)
+            null
+        else
+            EndpointFriendlyName(endpointParameters.friendlyName)
     }
 
     fun getNode(endpointEntityRepository: EndpointEntityRepository, nodeRequest: NodeRequest): Node? {
-        val splittedTopic = nodeRequest.nodeTopic.split("/")
-        return endpointEntityRepository.findByIdOrNull(splittedTopic[0])?.endpoint?.nodes?.get(splittedTopic[1])
+        val splitTopic = nodeRequest.nodeTopic.split("/")
+        return endpointEntityRepository.findByIdOrNull(splitTopic[0])?.endpoint?.nodes?.get(splitTopic[1])
     }
 
     fun getWotNode(endpointEntityRepository: EndpointEntityRepository, nodeRequest: NodeRequest, host: String): WotNode? {
+        val splitTopic = nodeRequest.nodeTopic.split("/")
```

Comment:
Check if the split operation gives at least 2 parts before accessing [0] and [1]

Reply:
Fixed in 0b262d6, same comment as above